.github/workflows/android.yml

```diff
@@ -0,0 +1,150 @@
+name: Android CI
+
+on:
+  push:
+    branches:
+      - master
+      - '!gh-pages'
+    paths:
+      # Rebuild when workflow configs change.
+      - .github/workflows/**
+      # Or when kotlin code changes.
+      - kotlin/**
+  pull_request:
+    paths:
+      - .github/workflows/**
+      - kotlin/**
+
+jobs:
+  assemble:
+    name: Assemble
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      # These setup steps should be common across all jobs in this workflow.
+      - uses: actions/checkout@v2
+      - uses: gradle/wrapper-validation-action@v1
+      - name: set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      ## Caching
+      - name: Cache Gradle wrapper
+        uses: actions/cache@v1
+        with:
+            path: ~/.gradle/wrapper
+            # Don't set restore-keys so cache is always only valid for the current build config.
+            key: gradle-wrapper-${{ runner.os }}-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}
+      - name: Cache build artifacts
+        uses: actions/cache@v1
+        with:
+            path: ~/.gradle/caches
+            # Include the SHA in the hash so this step always adds a cache entry. If we didn't use the SHA, the artifacts
+            # would only get cached once for each build config hash.
+            # Don't use ${{ runner.os }} in the key so we don't re-assemble for UI tests.
+            key: gradle-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle*') }}-${{ github.sha }}
+            # The first time a SHA is assembled, we still want to load dependencies from the cache.
+            # Note that none of jobs dependent on this one need restore keys, since they'll always have an exact hit.
+            restore-keys: |
```

Comment:
This is like science fiction. 😺

Reply:
I don't quite get what's going on w/the caching, but I don't see that any change is needed to the comments here. Just need to get myself up to speed.

Reply:
I'm not even sure caching is the right solution for per-run files. I think the cache is most useful for dependencies, things that don't change often across commits. But maybe we should use artifacts for communicating between jobs. Something we can play with after this merges.